net/udprelay/server.go

```diff
@@ -109,17 +166,17 @@ type ServerEndpoint struct {
 	// VNI (Virtual Network Identifier) is the Geneve header VNI the Server
 	// will use for transmitted packets, and expects for received packets
 	// associated with this endpoint.
-	VNI uint32
+	VNI ServerEndpointVNI
 
 	// BindLifetime is amount of time post-allocation the Server will consider
 	// the endpoint active while it has yet to be bound via 3-way bind handshake
 	// from both client parties.
-	BindLifetime time.Duration
+	BindLifetime ServerEndpointLifetime
 
 	// SteadyStateLifetime is the amount of time post 3-way bind handshake from
 	// both client parties the Server will consider the endpoint active lacking
 	// bidirectional data flow.
-	SteadyStateLifetime time.Duration
+	SteadyStateLifetime ServerEndpointLifetime
```

Comment:
The title of this PR suggests that we'd likely the JSON representation to be cross platform, given the lack of industry consensus on the right representation of a duration, perhaps we should express these instead as timestamps?

```go
// BindUntil is when the Server will consider the endpoint no longer active.
BindUntil time.Time

// SteadyStateUntil is when ...
SteadyStateUntil time.Time
```
Thus, these will be represented using RFC 3339 timestamps in JSON, which is very well agreed upon by all  platforms.

Reply:
There is a clock sync consideration for those fields. We tolerate transport over a network as negligible error for durations, but carrying timestamps would assume clocks are synchronized.

Reply:
I think time.Duration as time.Duration.String() is an acceptable middle ground.

I'll adjust the commit message to be more sensible once it uses the tstime type.